modules/lib/core/index.d.ts

```diff
@@ -142,6 +142,318 @@ export interface ScriptValue {
+    getHeader: RequestGetHeaderFunction
+}
+
+export interface DefaultRequest extends RequestInterface {
+    cookies: DefaultRequestCookies
+    headers: DefaultRequestHeaders
+}
+
+export type Request<
+    T extends Partial<RequestInterface> = Record<string, never>
+> = StrictMergeInterfaces<DefaultRequest, T>;
+
+export type SerializableRequest<T extends RequestInterface = DefaultRequest> = Omit<
+    Partial<T>,
+    'body' | 'contextPath' | 'rawPath' | 'repositoryId' | 'webSocket'
+> & {
+    body?: unknown[] | Record<string, unknown> | boolean | number | string | null;
+};
+
+export interface DefaultResponseHeaders extends ResponseHeaders {
+    'Cache-Control'?: string
+    'Content-Encoding'?: string
+    'Content-Type'?: string
+    'Content-Security-Policy'?: string
+    'Date'?: string
+    Etag?: string | number
+    Location?: string
+}
+
+// NOTE Even though PortalResponseSerializer allows non-array values,
+// when it comes back from Java the PortalResponseMapper will always return an array.
+// So perhaps it's better to enforce that, so it's consistent both ways.
+// It also causes problems in ResponseProcessors, since they work with both from and to Java.
+export interface PageContributions {
+	headBegin?: string[]
+	headEnd?: string[]
+	bodyBegin?: string[]
+	bodyEnd?: string[]
+}
+
+export type ResponseBody = unknown[] | Record<string, unknown> | boolean | number | string | null | ByteSource;
+
+export interface MappedResponse {
+    applyFilters: boolean
+    body?: ResponseBody
+    contentType: string
+    cookies: ResponseCookies
+    headers: ResponseHeaders
+    pageContributions: PageContributions
+    postProcess: boolean
+    status: number
+}
+
+export interface ResponseInterface extends Partial<MappedResponse> {
+    redirect?: string
+}
+
+export interface DefaultResponse extends ResponseInterface {
+    contentType?: LiteralUnion<'text/html' | 'application/json'>
+    headers?: DefaultResponseHeaders
+}
+
+export type Response<
+    T extends Partial<ResponseInterface> = Record<string, never>
+> = StrictMergeInterfaces<DefaultResponse, T>;
+
+export type RequestHandler<
+    RequestFromJava extends RequestInterface = DefaultRequest,
+    ResponseToJava extends ResponseInterface = DefaultResponse
+> = (request: RequestFromJava) => ResponseToJava;
+
+export type HttpFilterNext<
+    RequestToJava extends SerializableRequest = SerializableRequest<DefaultRequest>,
+    ResponseToJava extends ResponseInterface = DefaultResponse
+> = (request: RequestToJava) => ResponseToJava;
+
+export interface Controller {
+    all?: RequestHandler
+    // connect?: RequestHandler
+    delete?: RequestHandler
+    get?: RequestHandler
+    head?: RequestHandler
+    options?: RequestHandler
+    // patch?: RequestHandler
+    post?: RequestHandler
+    put?: RequestHandler
+    // trace?: RequestHandler
+    // TODO what about propfind, proppatch, mkcol, copy, move, lock and unlock?
```

Comment:
yes, what about them? I don't think they are supported out of the box

Reply:
I'm happy to remove them. They where mentioned in the previous pull request discussions.